src/dynsight/_internal/analysis/rdf.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from ovito.io import import_file
+from ovito.modifiers import (
+    CoordinationAnalysisModifier,
+    DeleteSelectedModifier,
+    ExpressionSelectionModifier,
+    LoadTrajectoryModifier,
+    TimeAveragingModifier,
+)
+from scipy.signal import find_peaks
+
+
+class RadialDistributionFunction:
+    """Object for computing and analyzing the Radial Distribution Function.
+
+    The Radial Distribution Function (RDF) describes how particle density
+    varies as a function of the distance.
+
+    Parameters:
+        trajectory:
+            List of file paths including trajectory and optional topology file.
+
+        pipeline:
+            OVITO pipeline used for importing and modifying the
+            trajectory data.
+
+        pair_distances:
+            Array of distances between particle pairs.
+
+        rdf:
+            Array of RDF values corresponding to the pair distances.
+
+        minima_points:
+            Tuple containing arrays of minima point distances and RDF values.
+    """
+
+    def __init__(
+        self,
+        trajectory_file: str,
+        topology_file: str | None = None,
+        xyz_cols: list[str] | None = None,
+    ) -> None:
+        """Initialize the RDF object and import the trajectory data.
```

Comment:
I do not think we should be importing the trajectory at this point because that comes at a (potentially large!) cost, and the user might prefer to initialize this calculator, and then run compute when they are ready.

Reply:
Move these parts to the compute method.